demo/dlb-init.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+ENABLED_DLB_PF_PCIIDS=${ENABLED_DLB_PF_PCIIDS:-270b 2710 2714}
+NUMVFS=${NUMVFS:-4}
+
+bind() {
+    BSF=$1
+    DRIVER=$2
+
+    LOOP=0
+    while [ "$LOOP" -lt  5 ]
+    do
+        if echo -n 2> /dev/null "$BSF" > /sys/bus/pci/drivers/"$DRIVER"/bind; then
+            break
+        else
+           LOOP=$(("$LOOP" + 1))
+           echo -n 2> /dev/null "$BSF" > /sys/bus/pci/devices/"$BSF"/driver/unbind
+            sleep 1
+        fi
+    done
+}
+
+unbind() {
+    BSF=$1
+    OLD_DRIVER=$2
+
+    LOOP=0
+    while [ "$LOOP" -lt  5 ]
+    do
+        if echo -n 2> /dev/null "$BSF" > /sys/bus/pci/drivers/"$OLD_DRIVER"/unbind;
+        then
+            break
+        else
+           LOOP=$(("$LOOP" + 1))
+            sleep 1
+        fi
+    done
+}
+
+#get dlb pf pci ids
+DEVS=$(for pf in $ENABLED_DLB_PF_PCIIDS; do lspci -n | grep -e "$pf" | grep -o -e "^\\S*"; done)
+for dev in $DEVS; do
+  DEVPATH="/sys/bus/pci/devices/0000:$dev"
+
+  #enable sriov_numvfs
+  unbind 0000:"$dev" dlb2
+  bind 0000:"$dev" dlb2
+  echo "$NUMVFS" | tee -a "$DEVPATH"/sriov_numvfs
+
+  for ((i=0;i<"$NUMVFS";i++));do
+    #unbind vf
+    vf=$(basename "$(realpath "$DEVPATH"/virtfn"$i")")
+    unbind "$vf" dlb2
+
+    #configure vf
+    echo $(($(cat "$DEVPATH"/total_resources/num_atomic_inflights) / "$NUMVFS")) | tee -a "$DEVPATH"/vf"${i}"_resources/num_atomic_inflights&&
```

Comment:
Consider writing this as a loop. Something like this:
```
for i in $(seq $(($NUMVFS-1)))
do
...
targets = "num_atomic_inflights num_dir_credits num_dir_ports ..."
for target in ${targets}
do
  val="$(cat ${DEVPATH}/total_resources/${target})
  tuned = $((${val} / ${NUMVFS}))
  echo $tuned | tee -a ${DEVPATH}/vf${i}_resources/${target}
done
...
done

```

Reply:
Ok, I will!! :)

Reply:
Actaully, I am curious why we do not set values for all files, but only some parts.
We do not set "num_cosX_ldb_ports". Why?
@bart0sh He may know the answer.?

Reply:
As far as I remember DLB devs suggested me what to set. However, that was long time ago, so you can try to set all possible resources.

Reply:
and/or read the latest release notes for that to configure.

Reply:
Well, "num_cosX_ldb_ports" existed from 2020, so maybe, it's not necessary thing to set.

@mythi what release note do you mean? I cannot find anything else than user guide inside the doc folder of dlb driver.

Maybe  [3.3.11. Class of service of this link](https://doc.dpdk.org/guides/eventdevs/dlb2.html) would be related, but I do not think it is something we need to consider now.!